File: suppet-frontend/src/app/commons/puppet-db/facts/puppet-db-fact-names.service.ts
import { Injectable } from '@angular/core';
import {AbstractPuppetDbService} from "../abstract-puppet-db.service";
import {HttpClient} from "@angular/common/http";

@Injectable({
  providedIn: 'root'
})
export class PuppetDbFactNamesService extends AbstractPuppetDbService {

  constructor(http: HttpClient) {
    super(http);
  }

  protected getBaseUrl(): string {
    return "/puppet-db/fact-names";
  }
}

File: suppet-backend/src/main/java/com/dyplom/suppet/service/puppetdb/facts/PuppetDbFactNamesService.java
package com.dyplom.suppet.service.puppetdb.facts;

import com.dyplom.suppet.service.puppetdb.AbstractPuppetDbService;
import org.springframework.stereotype.Service;

@Service
public class PuppetDbFactNamesService extends AbstractPuppetDbService {

    PuppetDbFactNamesService() {
        super("PuppetDbFactNamesService");
    }

    @Override
    protected String getBaseEndpointName() {
        return "fact-names";
    }
}
